qiskit/visualization/pulse/matplotlib.py

```diff
@@ -280,17 +280,27 @@ def __init__(self, style):
         """
         self.style = style or SchedStyle()
 
-    def _build_channels(self, schedule, channels_to_plot, t0, tf):
+    def _build_channels(self, schedule, channels_to_plot, t0, tf, framechange_channels=True):
         # prepare waveform channels
         drive_channels = collections.OrderedDict()
         measure_channels = collections.OrderedDict()
         control_channels = collections.OrderedDict()
         acquire_channels = collections.OrderedDict()
         snapshot_channels = collections.OrderedDict()
-
-        _channels = list(schedule.channels) + channels_to_plot
+        # take channels that do not only contain framechanges
+        if framechange_channels:
+            _channels = list(schedule.channels)
+        else:
+            _channels = []
```

Comment:
Using the `filter` method, we can get a version of this which is easier to read:
`_channels = schedule.filter(instruction_types=[FrameChange]).channels`

Reply:
i have a `set()` when i ran this command. I am wondering if the instruction_types` parameter supports also command type.

Reply:
It should be fine here that you get a `set` return type. The next line would be `_channels.update(channels_to_plot)`. Then, you can remove the line `_channels = list(set(_channels))` since sets are iterable, and it looks like `_channels` is only used in line 304 to iterate over.

`instructions_types` supports a list of `Command` types (subclasses of `Command`).

Reply:
Sorry i meant i had an empty set with `schedule.filter(instruction_types=[FrameChange]).channels` but i just realized that works with `FrameChangeInstruction` from `qiskit.pulse.commands.frame_change` module. Also in my code i take all channels and remove those with only framechanges,  i keep channels with framechange and other command. So i don't know if it's useful to use it at the beginning.

Reply:
right, sorry, it was supposed to be `FrameChangeInstruction`, not `FrameChange`. Hmm, would have been nice if that had raised an error message for you. I'll file an issue.

I see what you're saying about the filter -- you are right. what about this for readability:
```
channels = {}
for time, inst in schedule.instructions:
    if not isinstance(instruction, FrameChangeInstruction):
        channels.update(instruction.channels)
```

Reply:
then everywhere you can let `_channels` be a set